test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/CSharpStatementTest/StaticUsing_Complete_Spaced.stree.txt

```diff
@@ -1,13 +1,15 @@
-Directive block - Gen<None> - 40 - (0:0,0)
-    Transition span - Gen<None> - [@] - SpanEditHandler;Accepts:None - (0:0,0) - Tokens:1
-        SyntaxKind.Transition;[@];
-    Code span - Gen<Import:   static   global::System.Console;> - [using   static   global::System.Console] - SpanEditHandler;Accepts:AnyExceptNewline - (1:0,1) - Tokens:9
-        SyntaxKind.Keyword;[using];
-        SyntaxKind.Whitespace;[   ];
-        SyntaxKind.Keyword;[static];
-        SyntaxKind.Whitespace;[   ];
-        SyntaxKind.Identifier;[global];
-        SyntaxKind.DoubleColon;[::];
-        SyntaxKind.Identifier;[System];
-        SyntaxKind.Dot;[.];
-        SyntaxKind.Identifier;[Console];
+CSharpCodeBlock - [0..40)::40 - [@using   static   global::System.Console]
+    RazorDirective - [0..40)::40
```

Comment:
This is an interesting artifact. The directive became a 'code bock' instead of it's own thing.

Reply:
Yes. This is because when the HtmlParser encounters a single Transition(`@`), it doesn't parse further but calls `ParseBlock()` on the CSharp parser which returns a 'code block'. `CSharpCodeBlock` is usually a parent of anything CSharp. But I agree, we definitely need to clean this part up.